apps/quiz/views.py

```diff
@@ -9,15 +9,34 @@
 from apps.quiz.forms import QuizCreationForm
 from apps.quiz.quiz_factory import QuizFactory
 from apps.auth.permissions import IsAuthenticatedPermission
+from apps.quiz.models import Quiz
 
 
 bp = Blueprint('quiz', __name__)
 
 
 class QuizListView(BasePermissionCheckMethodView):
-
+    QUIZZES_PER_PAGE = 2
     template_name = "quiz/quiz_list.html"
 
+    def get_context(self) -> dict:
+        quizzes = Quiz.query.all()
+        max_page = int(len(quizzes) / self.QUIZZES_PER_PAGE)
+        page = request.args.get('page', default=1, type=int)
+
+        # Validate page number.
+        page = max(1, page)
```

Comment:
You are basically reinventing the wheel here, how about using this?
[https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/?highlight=paginate#flask_sqlalchemy.BaseQuery.paginate](https://flask-sqlalchemy.palletsprojects.com/en/2.x/api/?highlight=paginate#flask_sqlalchemy.BaseQuery.paginate)
or
[https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-ix-pagination](https://blog.miguelgrinberg.com/post/the-flask-mega-tutorial-part-ix-pagination)